src/libraries/TwabLib.sol

```diff
@@ -252,72 +256,92 @@ library TwabLib {
   }
 
   /**
-   * @notice Looks up a users balance at a specific time in the past.
+   * @notice Looks up a users balance at a specific time in the past. Ignores any updates during the current overwrite period.
    * @dev If the time is not an exact match of an observation, the balance is extrapolated using the previous observation.
-   * @dev Ensure timestamps are safe using isTimeSafe or by ensuring you're querying a multiple of the observation period intervals.
+   * @dev Ensure timestamps are safe using isDuringOverwritePeriod or by ensuring you're querying a multiple of the observation period intervals.
+   * @param PERIOD_LENGTH The length of an overwrite period
+   * @param PERIOD_OFFSET The offset of the first period
    * @param _observations The circular buffer of observations
    * @param _accountDetails The account details to query with
    * @param _targetTime The time to look up the balance at
    * @return balance The balance at the target time
    */
   function getBalanceAt(
+    uint32 PERIOD_LENGTH,
     uint32 PERIOD_OFFSET,
     ObservationLib.Observation[MAX_CARDINALITY] storage _observations,
     AccountDetails memory _accountDetails,
     uint32 _targetTime
   ) internal view returns (uint256) {
+    uint32 overwritePeriodStartTime = _overwritePeriodStartTime(PERIOD_LENGTH, PERIOD_OFFSET);
+    uint32 safeTargetTime = _targetTime < overwritePeriodStartTime ? _targetTime : overwritePeriodStartTime - 1;
     ObservationLib.Observation memory prevOrAtObservation = _getPreviousOrAtObservation(
       PERIOD_OFFSET,
       _observations,
       _accountDetails,
-      _targetTime
+      safeTargetTime
     );
     return prevOrAtObservation.balance;
   }
 
   /**
-   * @notice Looks up a users TWAB for a time range.
+   * @notice Looks up a users TWAB for a time range. Ignores any updates during the current overwrite period.
    * @dev If the timestamps in the range are not exact matches of observations, the balance is extrapolated using the previous observation.
-   * @dev Ensure timestamps are safe using isTimeRangeSafe.
+   * @param PERIOD_LENGTH The length of an overwrite period
+   * @param PERIOD_OFFSET The offset of the first period
    * @param _observations The circular buffer of observations
    * @param _accountDetails The account details to query with
    * @param _startTime The start of the time range
    * @param _endTime The end of the time range
    * @return twab The TWAB for the time range
    */
   function getTwabBetween(
+    uint32 PERIOD_LENGTH,
     uint32 PERIOD_OFFSET,
     ObservationLib.Observation[MAX_CARDINALITY] storage _observations,
     AccountDetails memory _accountDetails,
     uint32 _startTime,
     uint32 _endTime
   ) internal view returns (uint256) {
+    // The current period can still be changed; so the start of the period marks the beginning of unsafe timestamps.
+    uint32 overwritePeriodStartTime = _overwritePeriodStartTime(PERIOD_LENGTH, PERIOD_OFFSET);
```

Comment:
Whoops… realized this line should be passing the end time to compute the overwrite start time.  It shouldn’t be using the current time